### PR TITLE
fix: Drop support for retiring E-Commerce Service accounts

### LINKED ISSUE #36: Ecommerce retirement fails
# Summary

While running the retirement pipeline in an nutmeg installation that runs ecommerce and discovery I got the following error in the user retirement statuses:

```
Moved from RETIRING_ECOMMERCE (step 91) to ERRORED (step 191):
Client Error 404: https://{{ ECOMMERCE_HOST }}/api/v2/user/retire/
```

## My environment

* Tutor version: 14.2.3
* `tutor-contrib-retirement` version: 1.0.0
* Output of the retirement command/CronJob: `No users to retire`

## Additional context

A similar (pretty much the same) problem was reported back in 2020 in this forum post: https://discuss.openedx.org/t/retiring-a-learner-from-ecommerce/2951/2

I tried to investigate a little bit about that `/api/v2/user/retire/` endpoint but I wasn't able to find if it was ever implemented in ecommerce.

This endpoint is called from the [`EcommerceApi` client](https://github.com/openedx-unsupported/tubular/blob/fa0363c091f6a5f89a2bc5b5e0e9103592d33d8a/tubular/edx_api.py#L407) in the tubular repository.

I tried to check our old retirement pipelines but we didn't have the ecommerce steps.

I would like to confirm if this is a misconfiguration on my side or this have never really worked and it should be removed from the pipeline.

### PR DESCRIPTION
**Pull request by fghaas**
This never worked, and can't ever work, due to limitations in the E-Commerce service itself.

Thus, rather than rely on whether `ECOMMERCE_HOST` is defined to decide whether the retirement pipeline for that service should be enabled, always disable it.

Also, update the README explaining that this plugin can't be used to retire E-Commerce accounts.

Fixes #36.

### CODE REVIEW

**Comment by fghaas** [comment]
@mrtmm Please review this in the context of the discussion in #36. :)

**Comment by fghaas** [comment]
> Do you think it makes sense to allow for a custom pipeline, maybe via a tutor patch?

Not if it's just for the purpose of supporting ecommerce. The discussion in #36 makes me think people shouldn't be touching it with a ten-foot pole, so I am much more comfortable with the idea of dropping support for it altogether.

File: tutorretirement/templates/retirement/build/retirement/pipeline_config/config.yml [hunk 1]
```diff
@@ -3,13 +3,11 @@ client_secret: {{ RETIREMENT_EDX_OAUTH2_CLIENT_SECRET }}
 
 base_urls:
     lms: {{ "https" if ENABLE_HTTPS else "http" }}://{{ LMS_HOST }}
-    {% if ECOMMERCE_HOST is defined %}ecommerce: {{ "https" if ENABLE_HTTPS else "http" }}://{{ ECOMMERCE_HOST }}{% endif %}
```
Comment (bmtcril) [inline tutorretirement/templates/retirement/build/retirement/pipeline_config/config.yml:6]: I'm not sure how important this is to the community, but if you remove this it may impact the ability to retire all of the Segment IDs. I've got doubts if anyone in the community is using ecommerce + segment + tutor, but thought I'd mention it.

tl;dr: ecommerce can maintain a separate user id just for interacting with Segment. If the retirement config sets `fetch_ecommerce_segment_id` to True and includes the Segment retirement step, the scripts will query Ecommerce for this id and the segment_api will retire it along with the other things that get deleted from Segment. This is what the ecommerce retirement API that we were looking at in the other issue handles.

- https://github.com/openedx/edx-platform/blob/6d7fd926d151017d4fd2a502d7f47bc505f3e752/scripts/user_retirement/retire_one_learner.py#L176
- https://github.com/openedx/edx-platform/blob/6d7fd926d151017d4fd2a502d7f47bc505f3e752/scripts/user_retirement/utils/thirdparty_apis/segment_api.py#L18

**Review by mrtmm — APPROVED** [review]
Looks good to me, thanks!
I am not sure what to think/do about the Segment ID's mentioned by @bmtcril. I guess we can address the potential issues if/when reported by users?

File: tutorretirement/templates/retirement/build/retirement/pipeline_config/config.yml (hunk 1 above)
Comment (fghaas) [reply tutorretirement/templates/retirement/build/retirement/pipeline_config/config.yml:6]: @bmtcril We maintain a community Tutor plugin here. If ecommerce is, as you say, in such a state that a reasonable Open edX community deployment shouldn't be using it at all, and the retirement pipeline for it is in such a state that there is effectively no retirement for ecommerce at all, then I don't see any impetus for a community plugin to support it. Much less support something related to the integration of that thing with a third-party service.

I'd much prefer we explicitly *not* support it, so that people who are brave (reckless?) enough to use ecommerce (with or without Segment) know that they're on their own as far as retirement for those services is concerned.

Comment (bmtcril) [reply tutorretirement/templates/retirement/build/retirement/pipeline_config/config.yml:6]: Sounds good to me
